File: languages/go/entity/availability/publisher_args.go
package availability

import (
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
)

// PublisherArgs represents the arguments for a data publisher.
type PublisherArgs struct {
	// IpnsKey is a managed key or certificate object that will be used to create the IPNS record.
	IpnsKey *IpnsKey
}

func (e PublisherArgs) ToProto() *proto.PublisherArgs {
	var ipnsKey *proto.IpnsKey
	if e.IpnsKey != nil {
		ipnsKey = e.IpnsKey.ToProto()
	}

	return &proto.PublisherArgs{
		IpnsKey: ipnsKey,
	}
}

File: languages/js/src/entity/availability/publisher_args.ts
import * as proto from "../../bridge/proto/bloock_availability_entities";
import { IpnsKey } from "./ipns_key";

/**
 * Represents the arguments for a data publisher.
 */
export class PublisherArgs {
  /**
   * Is a unique identifier associated with the loader.
   */
  ipnsKey?: IpnsKey;

  /**
   * Constructs a PublisherArgs object with the specified parameters.
   * @param ipnsKey
   */
  constructor(ipnsKey?: IpnsKey) {
    if (ipnsKey) {
      this.ipnsKey = ipnsKey;
    }
  }

  public toProto(): proto.PublisherArgs {
    let ipnsKey: proto.IpnsKey | undefined;
    if (this.ipnsKey) {
      ipnsKey = this.ipnsKey.toProto();
    }

    return proto.PublisherArgs.fromPartial({ ipnsKey: ipnsKey });
  }
}
